app/chitter.rb

```diff
@@ -0,0 +1,105 @@
+require 'sinatra/base'
+require_relative 'data_mapper_setup'
+require_relative 'helper/current_user'
+require 'sinatra/flash'
+
+class Chitter < Sinatra::Base
+  include Helpers
+
+  use Rack::MethodOverride
+
+  register Sinatra::Flash
+  register Sinatra::Partial
+
+  enable :sessions, :static
+  enable :partial_underscores
+
+  set :session_secret, 'super secret'
+  set :partial_template_engine, :erb
+
+  get '/' do
+    erb :'index'
+  end
+
+  get '/users/new' do
+    @user = User.new
+    erb :'users/new'
+  end
+
+  post '/users' do
+    @user = User.new(name: params[:name],
+                     username: params[:username],
+                     email: params[:email],
+                     password: params[:password],
+                     password_confirmation: params[:password_confirmation])
+    if @user.save
+      session[:user_id] = @user.id
+      session[:username] = @user.username
+      redirect to('/')
+    else
+      flash.now[:errors] = @user.errors.full_messages
+      erb :'/users/new'
+    end
+  end
+
+  get '/sessions/new' do
+    @user = User.new
+    erb :'sessions/new'
+  end
+
+  post '/sessions/new' do
+    user = User.authenticate(params[:email], params[:password])
+    if user
+      session[:user_id] = user.id
+      session[:username] = user.username
+      redirect to('/')
```

Comment:
Do you need to put both username and user id in the session. Can you look the username up from the id?